src/Testcontainers/Builders/MTlsEndpointAuthenticationProvider.cs

```diff
@@ -0,0 +1,87 @@
+namespace DotNet.Testcontainers.Builders
+{
+  using System;
+  using System.IO;
+  using System.Linq;
+  using System.Security.Cryptography.X509Certificates;
+  using Docker.DotNet.X509;
+  using DotNet.Testcontainers.Configurations;
+  using Org.BouncyCastle.Crypto;
+  using Org.BouncyCastle.OpenSsl;
+  using Org.BouncyCastle.Pkcs;
+  using Org.BouncyCastle.Security;
+  using Org.BouncyCastle.X509;
+
+  /// <inheritdoc cref="IDockerRegistryAuthenticationProvider" />
+  internal sealed class MTlsEndpointAuthenticationProvider : TlsEndpointAuthenticationProvider
+  {
+    private static readonly X509CertificateParser CertificateParser = new X509CertificateParser();
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="MTlsEndpointAuthenticationProvider" /> class.
+    /// </summary>
+    /// <param name="customConfigurations">A list of custom configurations.</param>
+    public MTlsEndpointAuthenticationProvider(params ICustomConfiguration[] customConfigurations)
+      : base(customConfigurations)
+    {
+    }
+
+    /// <inheritdoc />
+    public override bool IsApplicable()
+    {
+      var certificatesFiles = new[] { ClientCertificateFileName, ClientCertificateKeyFileName };
+      return this.TlsEnabled && this.TlsVerifyEnabled && certificatesFiles.Select(fileName => Path.Combine(this.CertificatesDirectoryPath, fileName)).All(File.Exists);
```

Comment:
This is not correct. `this.TlsEnabled` can be `false`.
```suggestion
      return this.TlsVerifyEnabled && certificatesFiles.Select(fileName => Path.Combine(this.CertificatesDirectoryPath, fileName)).All(File.Exists);
```

Reply:
TLS verify sets TLS to `true` too.

Reply:
Ok, I did not catch that.

Reply:
You where right at the beginning. I changed that after your comment. Thanks.